extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/MicrosoftAuthServiceProvider.java

```diff
@@ -17,8 +20,18 @@ public String getServiceId() {
         return "microsoft";
     }
 
-    public AuthDataGenerator getAuthDataGenerator(String transferDataType) {
+    public AuthDataGenerator getAuthDataGenerator(String transferDataType, AuthMode mode) {
         // TODO
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public List<String> getImportTypes() {
+        return ImmutableList.of();
```

Comment:
Do we need to put values here and below?

Reply:
I left it blank since nothing was implemented yet for it but I didnt want to return null at the same time. I guess I could throw an Unsupported exception too. What makes the most sense?